plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -726,6 +727,15 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
                 setRollback(() -> deleteRecursivelyIfExists(new HdfsContext(session), hdfsEnvironment, deltaLogDirectory));
                 transactionLogWriter.flush();
             }
+            else {
+                if (!isCreateTableWithExistingLocationEnabled(session)) {
```

Comment:
Can we have a test doing this check ?

In the test it would be good to showcase how to address the issue via `delta.create-table-with-existing-location.enabled` session property and subsequently be able to actually create the table with a specified location.

Reply:
> Can we have a test doing this check ?

It's here
https://github.com/trinodb/trino/pull/14779/commits/ab5fb8186aabaa39ed4b83d9e98530d157be9789#diff-028acbd1056d8997d51304e1c0372aecfe3e4a82e9af3fe544e893e62f8bd616

> In the test it would be good to showcase how to address the issue via delta.create-table-with-existing-location.enabled session property and subsequently be able to actually create the table with a specified location.

It's here
https://github.com/trinodb/trino/pull/14779/commits/ab5fb8186aabaa39ed4b83d9e98530d157be9789#diff-a843105096de783ebf6404b76ef7f34b6a2d3267bed08508f4cba50d9033ace9

Reply:
Though I have to create a test using the session property. I will create it.

Reply:
Added below test cases to cover the legacy `CREATE TABLE ... WITH (LOCATION = ...)`

`TestDeltaLakeLegacyCreateTableWithExistingLocationUsingConfigProperty`
`TestDeltaLakeLegacyCreateTableWithExistingLocationUsingSessionProperty`
`TestDeltaLakeLegacyCreateTableFailureWithExistingLocation`